frontend/src/features/map_planning/store/MapStoreTypes.ts

```diff
@@ -97,47 +99,58 @@ export type History = Array<Action<unknown, unknown>>;
 export interface UntrackedMapSlice {
   untrackedState: UntrackedMapState;
   stageRef: React.RefObject<Konva.Stage>;
-  updateSelectedLayer: (selectedLayer: LayerName) => void;
-  updateLayerVisible: (layerName: LayerName, visible: UntrackedLayerState['visible']) => void;
-  updateLayerOpacity: (layerName: LayerName, opacity: UntrackedLayerState['opacity']) => void;
+  updateSelectedLayer: (selectedLayer: LayerDto) => void;
+  updateLayerVisible: (layerName: LayerType, visible: UntrackedLayerState['visible']) => void;
+  updateLayerOpacity: (layerName: LayerType, opacity: UntrackedLayerState['opacity']) => void;
   selectPlantForPlanting: (plant: PlantsSummaryDto | null) => void;
   selectPlanting: (planting: PlantingDto | null) => void;
 }
 
-/**
- * Utility array of the map layer's names.
- */
-export const LAYER_NAMES = [
-  'Base',
-  'Plant',
-  'Drawing',
-  'Dimension',
-  'Fertilization',
-  'Habitats',
-  'Hydrology',
-  'Infrastructure',
-  'Labels',
-  'Landscape',
-  'Paths',
-  'Shade',
-  'Soil',
-  'Terrain',
-  'Trees',
-  'Warnings',
-  'Winds',
-  'Zones',
-] as const;
+const LAYER_TYPES = Object.values(LayerType);
+
+export const TRACKED_DEFAULT_STATE: TrackedMapState = {
+  layers: {
```

Comment:
I think we should refactor this to

```
  layers: LAYER_TYPES.reduce(
      (acc, layerName) => {
        acc[layerName]: {
          index: layerName,
          objects: [],
        },
        return acc
      },
      {} as TrackedLayers,
    ),
```
The spread operator can quickly slow down the code when not used carefully. The reason is that the spread operator has to iterate over the list every time it gets called.
When we mutate the variable instead we save a lot of iterations. We also don't have to be worried about side effects because the variable acc is scoped to the reduce function.

I don't think it is too much of an issue here because the size of the elements is rather small but if this is executed often it still matters and I think to try to write efficient code is still good practice.

Reply:
@Bushuo can you look into this? Doesn't seem to work without some modifications and I am not really familiar with the notation.